### PR TITLE
[GR-34703] Sealed classes support

### LINKED ISSUE #3870: Runtime crash in spring 5.3/spring-native with JDK 17
**Describe the issue**
The generated image crashes at runtime when `isSealed` method of `Class` is invoked through reflection. Happens at https://github.com/spring-projects/spring-framework/blob/5.3.x/spring-aop/src/main/java/org/springframework/aop/framework/AopProxyUtils.java#L145

**Describe GraalVM and your environment:**
 - GraalVM CE 21.3.0-dev-20211001_0733
 - JDK **17**
 - OS: **Linux**
 - Architecture: **AMD64**

**More details**
```
Caused by: com.oracle.svm.core.jdk.UnsupportedFeatureError: Unsupported method java.lang.Class.isSealed() is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
	at com.oracle.svm.core.util.VMError.unsupportedFeature(VMError.java:87)
	at java.lang.Class.isSealed(DynamicHub.java:4563)
	at java.lang.reflect.Method.invoke(Method.java:568)
	at org.springframework.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:282)
	at org.springframework.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:266)
	at org.springframework.aop.framework.AopProxyUtils.completeProxiedInterfaces(AopProxyUtils.java:145)
```

### PR DESCRIPTION
**Pull request by graalvmbot**
Implements sealed classes support for JDK17+
Fixes #3870

### CODE REVIEW

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SealedClassSupport.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    @Override
+    public Class<?>[] getPermittedSubclasses(Class<?> clazz) {
+        return null;
+    }
+}
+
+@AutomaticFeature
+final class SealedClassFeatureBeforeJDK17 implements Feature {
```
Comment (peter-hofer) [inline substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SealedClassSupport.java:72]: Should be called `SealedClassFeatureJDK11OrEarlier` for consistency.

File: substratevm/src/com.oracle.svm.configure/src/com/oracle/svm/configure/trace/ReflectionProcessor.java [hunk 2]
```diff
@@ -156,6 +156,11 @@ public void processEntry(Map<String, ?> entry) {
                 configuration.getOrCreateType(condition, clazz).setAllDeclaredClasses();
                 break;
             }
+            case "getPermittedSubclasses":
+            case "isSealed": {
```
Comment (peter-hofer) [inline substratevm/src/com.oracle.svm.configure/src/com/oracle/svm/configure/trace/ReflectionProcessor.java:160]: Nothing seems to generate a trace event for `isSealed`, please remove.

**Review by peter-hofer — APPROVED** [review]

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SealedClassSupport.java (hunk 1 above)
Comment (lazar-mitrovic) [reply substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SealedClassSupport.java:72]: I took the naming scheme from `RecordFeatureBeforeJDK17`, should we change that as well or leave it as is?

Comment (peter-hofer) [reply substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SealedClassSupport.java]: I have the same concerns, you can change it if you'd like.

**Pull request merged**